src/lib/support/CHIPMem-Malloc.cpp

```diff
@@ -26,42 +26,90 @@
 #include <core/CHIPConfig.h>
 #include <support/CHIPMem.h>
 
-#include <string.h>
+#include <stdlib.h>
+
+#ifndef NDEBUG
+#include <stdio.h>
+#endif
 
 #if CHIP_CONFIG_MEMORY_MGMT_MALLOC
 
 namespace chip {
 namespace Platform {
 
+#ifdef NDEBUG
+
+#define VERIFY_INITIALIZED()
+
+#else
+
+static int memoryInitialized = 0;
```

Comment:
Shouldn't this be atomic in builds w/ locking?

Reply:
Yes, good point. And should also check against calling Init while Init'd and Shutdown while shut down.